adapters/33across/33across.go

```diff
@@ -81,63 +104,84 @@ func (a *TtxAdapter) makeRequest(request *openrtb.BidRequest) (*adapters.Request
 		Uri:     a.endpoint,
 		Body:    reqJSON,
 		Headers: headers,
-	}, errs
+	}, nil
 }
 
-// Mutate the request to get it ready to send to ttx.
-func preprocess(request *openrtb.BidRequest) error {
-	var imp = &request.Imp[0]
+func makeImps(imp openrtb.Imp) (openrtb.Imp, error) {
+	if imp.Banner == nil && imp.Video == nil {
+		return openrtb.Imp{}, &errortypes.BadInput{
+			Message: fmt.Sprintf("Imp ID %s must have at least one of [Banner, Video] defined", imp.ID),
+		}
+	}
+
 	var bidderExt adapters.ExtImpBidder
 	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
-		return &errortypes.BadInput{
+		return openrtb.Imp{}, &errortypes.BadInput{
```

Comment:
Why not just return an imp pointer in this function and accordingly return `nil` here and below when you encounter an error? You can then just dereference it above when appending it to the imp array. Right now, we're creating these empty imp instances which aren't really gonna be used and will just take up memory until they are garbage collected

Reply:
Ah gotcha, wasn't sure how to return nil values with structs (sorry, new to Golang). Will do this.

Reply:
No worries at all :) Thanks for addressing it